libmamba/src/core/transaction.cpp

```diff
@@ -670,17 +674,18 @@ namespace mamba
             solver.install_specs()
         );
 
-        if (m_transaction_context.relink_noarch && pool->installed != nullptr)
+        ::Pool* const pool_ptr = m_pool;
```

Comment:
What's the reason for having another pool here `pool_ptr`?

Reply:
it's not a copy, `m_pool` is a `MPool` and `pool_ptr` a `::Pool*`, due to `MPool::operator ::Pool*`.
A static cast would be more explicit...